Added the ability to mark ignored players inside clan chat

melkypie opened the pull request:
I will reference #10515 although it does not fix the issue but only fixes the second idea of clan chats marking ignores.

melkypie commented:
Took me some commits to show what I changed in the script and added some fixes suggested in discord.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java:
```diff
@@ -541,6 +542,13 @@ public void onScriptCallbackEvent(ScriptCallbackEvent scriptCallbackEvent)
 				break;
 			}
+			case "clanChatChannelRebuild":
+				if (!config.showIgnores())
```
```suggestion
				if (config.showIgnores())
				{
					clientThread.invoke(this::colorIgnoredPlayers);
				}
```

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java:
```diff
@@ -668,4 +676,23 @@ private void confirmKickPlayer(final String kickPlayerName)
+
+	private void colorIgnoredPlayers()
+	{
+		if (client.getWidget(WidgetInfo.CLAN_CHAT_LIST).getChildren() == null)
```
```suggestion
		Widget clanChatList = client.getWidget(WidgetInfo.CLAN_CHAT_LIST)
		if (clanChatList == null || clanChatList.getChildren() == null)
```

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java:
```diff
@@ -668,4 +676,23 @@ private void confirmKickPlayer(final String kickPlayerName)
+		}
+
+		Color ignoreColor = config.showIgnores() ? config.showIgnoresColor() : Color.WHITE;
+		for (Widget listWidget : client.getWidget(WidgetInfo.CLAN_CHAT_LIST).getChildren())
```
```suggestion
		for (Widget listWidget : clanChatList.getChildren())
```

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java:
```diff
@@ -668,4 +676,23 @@ private void confirmKickPlayer(final String kickPlayerName)
+		Color ignoreColor = config.showIgnores() ? config.showIgnoresColor() : Color.WHITE;
+		for (Widget listWidget : client.getWidget(WidgetInfo.CLAN_CHAT_LIST).getChildren())
+		{
+			if (listWidget.getText().isEmpty() || listWidget.getText().matches("World \\d+") ||
```
Is there a better way to detect this? Some people have an rsn matching `World \\d+`

melkypie replied:
Added so it hops every 3 widgets, since the order that the widgets go in is name, world, icon.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatConfig.java:
```diff
@@ -148,4 +149,26 @@ default boolean confirmKicks()
+
+	@ConfigItem(
+		keyName = "showIgnores",
+		name = "Marks players who are on ignore list",
```
Ideally the config names should be fairly short since it is what shows in the panel, and then have a longer description. maybe `Mark ignored players` or something.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java:
```diff
@@ -167,6 +167,7 @@ public void onConfigChanged(ConfigChanged configChanged)
 			{
 				removeClanCounter();
 			}
+			colorIgnoredPlayers();
```
newline above this.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java:
```diff
@@ -668,4 +675,26 @@ private void confirmKickPlayer(final String kickPlayerName)
+		for (int i = 0; i < clanChatList.getChildren().length; i += 3)
+		{
+			Widget listWidget = clanChatList.getChild(i);
+			if (listWidget.getText().isEmpty() || client.getIgnoreContainer().findByName(listWidget.getText()) == null)
```
You can pull the ignore container out of this loop

melkypie commented:
Made some optimizations, instead of scripts it will be using the same algorithm that gets clan member count and added the requested changes

melkypie commented:
So I reverted the changes as requested and added the requested changes. Also found out that using getDynamicChildren on the CLAN_CHAT_LIST widget I could get the player name widget so I remove the every 3 widgets loop and replaced it with a for each dynamic widget

melkypie commented:
Reverted back to every 3x widgets

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java:
```diff
@@ -167,6 +169,8 @@ public void onConfigChanged(ConfigChanged configChanged)
 				removeClanCounter();
 			}
+
+			colorIgnoredPlayers();
```
This needs to run on the client thread. Also you could check the config key being changed.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatConfig.java:
```diff
@@ -148,4 +149,26 @@ default boolean confirmKicks()
+	@ConfigItem(
+		keyName = "showIgnores",
+		name = "Marks ignored players",
+		description = "Allows you to pick out players that are from your ignore list",
```
maybe "Recolors players on your ignore list" or something. "Allows you to pick out" could mean anything.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java:
```diff
@@ -668,4 +678,28 @@ private void confirmKickPlayer(final String kickPlayerName)
+		Color ignoreColor = config.showIgnores() ? config.showIgnoresColor() : Color.WHITE;
+		NameableContainer<Ignore> ignoreContainer = client.getIgnoreContainer();
+		// Iterate every 3 widgets, since the order of widgets is name, world, icon
+		// and World \\d+ could be a players name
```
this comment is just explaining why your previous implementation is a bad idea, but has nothing to do with this. i would remove it.

melkypie commented:
Applied the requested changes and also it does immediately change after you enable/disable the config but didn't after disabling the entire plugin, so I have fixed that.
